src/inmanta/agent/local_executor.py

```diff
@@ -0,0 +1,66 @@
+"""
+    Copyright 2024 Inmanta
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    Contact: code@inmanta.com
+"""
+
+import os
+import typing
+from concurrent.futures import ThreadPoolExecutor
+
+import inmanta.agent.executor as executor
+from inmanta.agent import config as cfg
+from inmanta.loader import ModuleSource
+
+
+class LocalExecutor(executor.Executor):
```

Comment:
I do not entirely understand where term `Local` comes from here?

Reply:
It is the one that runs in the agent process itself.

Reply:
Shouldn't we call it `InCurrentProcessExecutor` instead? It's quite a long name, but it's clear. `Local` seems to imply that the other executors run on a remote machine, which is not really the case.